default.nix

```diff
@@ -205,10 +210,11 @@ in lib.fix (self: {
       struct' = name: def: typedef' {
         inherit name def;
         checkType = value: if isAttrs value
-          then (checkStruct (self.attrs self.type def) value)
+          then if def ? _ then (checkStruct (removeAttrs (self.attrs self.type def) [ "_" ]) value def._)
```

Comment:
I'm wary of the `?` operator here because of its mysterious type-casting properties, could you go for `hasAttr "_" def`?

Reply:
Sure, I'll get this changed. I wasn't aware of `?` being weird, do you have an example?